components/script/dom/element.rs

```diff
@@ -680,6 +680,34 @@ impl Element {
         Ok(shadow_root)
     }
 
+    /// Attach a UA widget shadow root with its default parameters.
+    /// Additionally mark ShadowRoot to use styling configuration for a UA widget.
+    ///
+    /// TODO: Ideally, all of the UA shadow root should use UA widget styling, but
+    ///       some of the UA widget implemented prior to the implementation of Gecko's
+    ///       UA widget matching might need some tweaking.
+    pub(crate) fn attach_ua_shadow_root(
+        &self,
+        use_ua_widget_styling: bool,
+        can_gc: CanGc,
+    ) -> DomRoot<ShadowRoot> {
+        let root = self
+            .attach_shadow(
+                IsUserAgentWidget::Yes,
+                ShadowRootMode::Closed,
+                false,
+                false,
+                true,
```

Comment:
This sets `delegates_focus = true`, but previously all cases were using `false`. Is the change deliberate?
Also please add some comment for the bool parameters so that a reader can know what they are doing.

Reply:
After further thoughts, it seems much more reasonable. We would need to implement more advanced element focus to be able to support delegate focus properly. Thanks!